src/panels/lovelace/editor/config-elements/hui-picture-glance-card-editor.ts

```diff
@@ -0,0 +1,242 @@
+import "@polymer/paper-listbox/paper-listbox";
+
+import "../../components/hui-action-editor";
+import "../../components/hui-entity-editor";
+import "../../../../components/entity/ha-entity-picker";
+
+import { struct } from "../../common/structs/struct";
+import {
+  EntitiesEditorEvent,
+  EditorTarget,
+  actionConfigStruct,
+} from "../types";
+import { HomeAssistant } from "../../../../types";
+import { LovelaceCardEditor } from "../../types";
+import { fireEvent } from "../../../../common/dom/fire_event";
+import { configElementStyle } from "./config-elements-style";
+import { ActionConfig } from "../../../../data/lovelace";
+import { PictureGlanceCardConfig } from "../../cards/types";
+import { EntityConfig } from "../../entity-rows/types";
+import { processEditorEntities } from "../process-editor-entities";
+
+const entitiesConfigStruct = struct.union([
+  {
+    entity: "entity-id",
+    name: "string?",
+    icon: "icon?",
+  },
+  "entity-id",
+]);
+
+const cardConfigStruct = struct({
+  type: "string",
+  title: "string?",
+  entity: "string?",
+  image: "string?",
+  camera_image: "string?",
+  camera_view: "string?",
+  state_iamge: "object?",
+  aspect_ratio: "string?",
+  tap_action: struct.optional(actionConfigStruct),
+  hold_action: struct.optional(actionConfigStruct),
+  entities: [entitiesConfigStruct],
+});
+
+@customElement("hui-picture-glance-card-editor")
+export class HuiPictureGlanceCardEditor extends LitElement
+  implements LovelaceCardEditor {
+  @property() public hass?: HomeAssistant;
+
+  @property() private _config?: PictureGlanceCardConfig;
+
+  @property() private _configEntities?: EntityConfig[];
+
+  public setConfig(config: PictureGlanceCardConfig): void {
+    config = cardConfigStruct(config);
+    this._config = config;
+    this._configEntities = processEditorEntities(config.entities);
+  }
+
+  get _entity(): string {
+    return this._config!.entity || "";
+  }
+
+  get _title(): string {
+    return this._config!.title || "";
+  }
+
+  get _image(): string {
+    return this._config!.image || "";
+  }
+
+  get _camera_image(): string {
+    return this._config!.camera_image || "";
+  }
+
+  get _camera_view(): string {
+    return this._config!.camera_view || "";
+  }
+
+  get _state_image(): {} {
+    return this._config!.state_image || {};
+  }
+
+  get _aspect_ratio(): string {
+    return this._config!.aspect_ratio || "";
+  }
+
+  get _tap_action(): ActionConfig {
+    return this._config!.tap_action || { action: "none" };
+  }
+
+  get _hold_action(): ActionConfig {
+    return this._config!.hold_action || { action: "none" };
+  }
+
+  get _show_name(): boolean {
+    return this._config!.show_name || false;
+  }
+
+  get _show_state(): boolean {
+    return this._config!.show_state || false;
+  }
+
+  protected render(): TemplateResult | void {
+    if (!this.hass) {
+      return html``;
+    }
+
+    const actions = ["more-info", "toggle", "navigate", "call-service", "none"];
```

Comment:
shouldn't this be exported from `hui-action-editor` or are there actions specific to the card?

Reply:
Not all cards support the same actions. Off the top of my head I remember that picture card doesn't have more-info